Stratis.Bitcoin/BlockPulling/StoreBlockPuller.cs

```diff
@@ -25,14 +24,16 @@ public StoreBlockPuller(ConcurrentChain chain, Connection.IConnectionManager nod
         }
 
         /// <summary>
-        /// Prepares and invokes download task for a single block.
+        /// Prepares and invokes a download task for mulitple blocks.
         /// </summary>
-        /// <param name="downloadRequest">Description of a block to download.</param>
-        public void AskBlock(ChainedBlock downloadRequest)
+        public void AskForMultipleBlocks(ChainedBlock[] downloadRequests)
         {
-            this.logger.LogTrace("({0}:'{1}/{2}')", nameof(downloadRequest), downloadRequest.HashBlock, downloadRequest.Height);
+            foreach (var downloadRequest in downloadRequests)
+            {
+                this.logger.LogTrace("({0}:'{1}/{2}')", nameof(downloadRequest), downloadRequest.HashBlock, downloadRequest.Height);
+            }
```

Comment:
`()` should appear only once in a log for each method call. So does `(-)`
This is better to be replaced with 
```
 this.logger.LogTrace("({0}.{1}:{2}')", nameof(downloadRequest), nameof(downloadRequest.Count), downloadRequest.Count);
```

Note that you are calling `BlockPuller.AskBlocks` which does enumerate block heights, so the count here is enough.